lib/web/apiserver.go

```diff
@@ -245,14 +245,12 @@ func (h *APIHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	// request is already authenticated (has a session cookie), forward to
 	// application handlers. If the request is unauthenticated and requesting a
 	// FQDN that is not of the proxy, redirect to application launcher.
+
 	if h.appHandler != nil && (app.HasFragment(r) || app.HasSession(r) || app.HasClientCert(r)) {
 		h.appHandler.ServeHTTP(w, r)
 		return
 	}
-	if redir, ok := app.HasName(r, h.handler.cfg.ProxyPublicAddrs); ok {
```

Comment:
Why is this redirect no longer needed? Does it still work if you try to access an application over its "custom" public address that doesn't match the proxy public addr?

Reply:
My bad, I think I had removed this originally when trying something else out and forgot to add it back in. Added.